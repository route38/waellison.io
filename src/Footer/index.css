#siteFooter {
  grid-row-start: 3;
  grid-row-end: 4;
  grid-column-start: 1;
  grid-column-end: 6;
  box-sizing: border-box;
  padding: 0.5rem 2rem;
  color: var(--foreground-dark);
  background-color: var(--background-black);
  display: flex;
  align-items: center;
}

#siteFooter ul li {
  margin-right: 0;
}

.footer-nav li:not(:last-child)::after {
  content: " \2726 ";
}
