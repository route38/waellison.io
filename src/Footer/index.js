import "./index.css";
import { NavLink } from "react-router-dom";

const Footer = () => {
    return (
        <footer id="siteFooter">
            <p>Copyright © 2022 by William Ellison. • <NavLink className="navLink" to="/colophon">Colophon</NavLink></p>
        </footer>
    );
}

export default Footer;