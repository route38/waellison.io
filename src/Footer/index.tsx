import React from "react";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { solid, regular, brands } from "@fortawesome/fontawesome-svg-core/import.macro";

const Footer = () => {
  return (
    <footer id="siteFooter">
      <div className="meta">
        <p>Copyright Â© 2022 by William Ellison.</p>
      </div>
      <div className="socialBox">
        <a href="//github.com/waellison" className="socialBtn githubLink">
            <FontAwesomeIcon icon={brands("github-alt")} />
        </a>
        <a href="//linkedin.com/in/tnwae" className="socialBtn linkedinLink">
            <FontAwesomeIcon icon={brands("linkedin-in")}/>
        </a>
        <a href="//twitter.com/waedotpy" className="socialBtn twitterLink">
            <FontAwesomeIcon icon={brands("twitter")}/>
        </a>
        <a href="//instagram.com/nobettershoulder" className="socialBtn instaLink">
            <FontAwesomeIcon icon={brands("instagram")}/>
        </a>
        <a href="mailto:waellison@gmail.com" className="socialBtn emailLink">
            <FontAwesomeIcon icon={solid("envelope")}/>
        </a>
        <a href="//waellison.blog" className="socialBtn blogLink">
            <FontAwesomeIcon icon={solid("globe-americas")}/>
        </a>
      </div>
    </footer>
  );
};

export default Footer;
