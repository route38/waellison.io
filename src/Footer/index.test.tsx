import React from "react";
import renderer from "react-test-renderer";
import Footer from "./index";

it("renders correct footer component", () => {
  const component: renderer.ReactTestRenderer | null = renderer.create(<Footer />);
  const tree: renderer.ReactTestRendererJSON | renderer.ReactTestRendererJSON[] | null = component.toJSON();
  expect(tree).toMatchSnapshot();
});
