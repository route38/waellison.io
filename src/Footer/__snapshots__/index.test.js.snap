// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct footer component 1`] = `
<footer
  id="siteFooter"
>
  <p>
    Copyright Â© 2022 by William Ellison.
  </p>
</footer>
`;
